geoportailv3/static/js/locationinfo/locationinfodirective.js

```diff
@@ -0,0 +1,278 @@
+        }
+      }, this);
+    }
+  }, this));
+
+  this['map'].getViewport()
+    .addEventListener('contextmenu', goog.bind(function(event) {
+        event.preventDefault(); // disable right-click menu on browsers
+      }, this));
+
+  /**
+   * @type {angular.$timeout}
+   * @private
+   */
+  this.$timeout_ = $timeout;
+
+  /**
+   * @type {?Object<number, number>}
+   * @private
+   */
+  this.startPixel_ = null;
+
+  /**
+   * @type {angular.$q.Promise|undefined}
+   * @private
+   */
+  this.holdPromise_ = undefined;
+
+  this['map'].on('pointerdown', goog.bind(function(event) {
+    if (event.originalEvent.which === 3) { // if left mouse click
+      this.showInfoPane_(event.originalEvent);
+    } else if (!(event.originalEvent instanceof MouseEvent)) {
+      // if touch input device
+      this.$timeout_.cancel(this.holdPromise_);
+      this.startPixel_ = event.pixel;
+      var that = this;
+      this.holdPromise_ = this.$timeout_(function() {
+        that.showInfoPane_(event.originalEvent);
+      }, 500, false);
+    }
+  }, this));
+  this['map'].on('pointerup', goog.bind(function(event) {
+    this.$timeout_.cancel(this.holdPromise_);
+    this.startPixel_ = null;
+  }, this));
+  this['map'].on('pointermove', goog.bind(function(event) {
+    if (this.startPixel_) {
+      var pixel = event.pixel;
+      var deltaX = Math.abs(this.startPixel_[0] - pixel[0]);
+      var deltaY = Math.abs(this.startPixel_[1] - pixel[1]);
+      if (deltaX + deltaY > 6) {
+        this.$timeout_.cancel(this.holdPromise_);
+        this.startPixel_ = null;
+      }
+    }
+  }, this));
+};
+
+
+/**
+ * @param {MouseEvent|TouchEvent} event
+ * @private
+ */
+app.LocationinfoController.prototype.showInfoPane_ =
+    function(event) {
+  event.preventDefault();
+  this['open'] = true;
+  this['appSelector'] = 'locationinfo';
+  var clickCoordinate = this.map.getEventCoordinate(event);
+  this['coordinate'] = clickCoordinate;
+  var feature = /** @type {ol.Feature} */
+      (new ol.Feature(new ol.geom.Point(clickCoordinate)));
+  var features = this.featureOverlay_.getFeatures();
+  features.clear();
+  features.push(feature);
+  this['featureOverlayItemExists'] = true;
+  this.getElevation_(clickCoordinate).then(goog.bind(
+      function(elevation) {
+        this['elevation'] = elevation;
+      }, this
+      ));
+  this.getShorturl_(clickCoordinate).then(goog.bind(
+      function(shorturl) {
+        this['url'] = shorturl;
+        this['qrUrl'] = this.qrServiceUrl_ + '?url=' + shorturl;
+      }, this));
+
+};
+
+
+/**
+ * @return {ol.FeatureOverlay} The feature overlay.
+ * @private
+ */
+app.LocationinfoController.prototype.createFeatureOverlay_ =
+    function() {
+  var featureFill = new ol.style.Fill({
+    color: [255, 255, 0, 0.6]
+  });
+  var featureStroke = new ol.style.Stroke({
+    color: [255, 155, 55, 1],
+    width: 3
+  });
+  var featureOverlay = new ol.FeatureOverlay({
```

Comment:
`ol.FeatureOverlay` has been removed from OpenLayers 3, but we probably need to globally address that issue.